src/components/validation-form/index.tsx

```diff
@@ -0,0 +1,190 @@
+/* eslint-disable no-console */
+import React, { memo, FC, useState, FormEventHandler } from 'react';
+
+import { compose } from 'redux';
+
+import Button from '@fellesdatakatalog/button';
+
+import { ExpansionPanelBody, ExpansionPanelHead } from '../expansion-panel';
+import GraphFieldset from '../graph-fieldset';
+import Translation from '../translation';
+
+import SC from './styled';
+
+import type { ShapesCollection, ValidationRequest } from '../../types';
+import { InputType } from '../../types/enums';
+import Checkbox, { LabelPosition } from '../checkbox';
+
+interface ExternalProps {
+  dataGraph?: File | string;
+  shapesGraph?: File | string;
+  shapesCollection: ShapesCollection | null;
+  isLoading: boolean;
+  onValidate: (request: ValidationRequest) => void;
+}
+
+interface Props extends ExternalProps {}
+
+const ValidationForm: FC<Props> = ({
+  dataGraph,
+  shapesGraph,
+  shapesCollection,
+  isLoading,
+  onValidate,
+  ...props
+}) => {
+  const shapesOptions = shapesCollection
+    ? shapesCollection.shapes.map(shapesDef => ({
+        value: shapesDef.url,
+        label: `${shapesDef.specificationName} ${shapesDef.specificationVersion}`
+      }))
+    : [];
+
+  const defaultShapes = shapesCollection?.shapes.find(
+    shapesDef =>
+      shapesDef.specificationName === 'DCAT-AP-NO' &&
+      shapesDef.specificationVersion === '2.0'
+  )?.url;
+
+  const [inputDataGraph, setInputDataGraph] = useState<File | string>(
+    dataGraph ?? ''
+  );
+  const [inputShapesGraph, setInputShapesGraph] = useState(shapesGraph ?? '');
+  const [inputConfigExpand, setInputConfigExpand] = useState(true);
+  const [
+    inputConfigIncludeExpandedTriples,
+    setInputConfigIncludeExpandedTriples
+  ] = useState(false);
+
+  if (!inputShapesGraph && defaultShapes) {
+    setInputShapesGraph(defaultShapes);
+  }
+
+  const validateInput = () => {
+    if (inputDataGraph) {
+      onValidate({
+        dataGraph: inputDataGraph,
+        shapesGraph: inputShapesGraph,
+        config: {
+          expand: inputConfigExpand,
+          includeExpandedTriples: inputConfigIncludeExpandedTriples
+        }
+      });
+    }
+  };
+
+  const handleSubmitForm: FormEventHandler = e => {
+    e.preventDefault();
+    validateInput();
+  };
+
+  const handleOnChangeDataGraph = (
+    inputType: InputType,
+    graph: File | string
+  ) => {
+    if (inputType) {
+      setInputDataGraph(graph);
+    }
+  };
+
+  const handleOnChangeShapesGraph = (
+    inputType: InputType,
+    graph: File | string
+  ) => {
+    if (inputType) {
+      setInputShapesGraph(graph);
+    }
+  };
+
+  return (
+    <SC.ValidationInputForm {...props} onSubmit={handleSubmitForm}>
```

Comment:
Her kaller det for integrasjon med formik

Reply:
why? liker ikke formik...